src/core/load-fragment.c

```diff
@@ -3181,6 +3181,86 @@ int config_parse_syscall_filter(
         }
 }
 
+int config_parse_syscall_log(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        ExecContext *c = data;
+        _unused_ const Unit *u = userdata;
+        bool invert = false;
+        const char *p;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(u);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                c->syscall_log = hashmap_free(c->syscall_log);
+                c->syscall_log_allow_list = false;
+                return 0;
+        }
+
+        if (rvalue[0] == '~') {
+                invert = true;
+                rvalue++;
+        }
+
+        if (!c->syscall_log) {
+                c->syscall_log = hashmap_new(NULL);
+                if (!c->syscall_log)
+                        return log_oom();
+
+                if (invert)
+                        /* Log everything but the ones listed */
+                        c->syscall_log_allow_list = false;
+                else
+                        /* Log nothing but the ones listed */
+                        c->syscall_log_allow_list = true;
```

Comment:
```c
c->syscall_log_allow_list = !invert;
```
?

Reply:
While shorter, I think it would make the intent less obvious. Also the longer version matches `config_parse_syscall_filter()` (since it was copied from there) but the comments show the difference between "allow" and "log" actions when talking about allow-lists or deny-lists. Of course the comments could be retained somehow with the short version.

Reply:
The compiler generates the same code without branches, except for -O0, where shorter version produces better code ([godbolt](https://godbolt.org/z/eWj6GP)).